tests/unit/FirebaseStorage.spec.js

```diff
@@ -0,0 +1,304 @@
+'use strict';
+describe('$firebaseStorage', function () {
+  var $firebaseStorage;
+  var URL = 'https://oss-test.firebaseio.com';
+
+  beforeEach(function () {
+    module('firebase.storage');
+  });
+
+  describe('<constructor>', function () {
+
+    var $firebaseStorage;
+    var $q;
+    var $rootScope;
+    var $firebaseUtils;
+    beforeEach(function () {
+      module('firebase.storage');
+      inject(function (_$firebaseStorage_, _$q_, _$rootScope_, _$firebaseUtils_) {
+        $firebaseStorage = _$firebaseStorage_;
+        $q = _$q_;
+        $rootScope = _$rootScope_;
+        $firebaseUtils = _$firebaseUtils_;
+      });
+    });
+
+    function setupPutTests(file, mockTask, isPutString) {
+      var ref = firebase.storage().ref('thing');
+      var task = null;
+      var storage = $firebaseStorage(ref);
+      var putMethod = isPutString ? 'putString': 'put';
+      // If a MockTask is provided use it as the
+      // return value of the spy on put
+      if (mockTask) {
+        spyOn(ref, putMethod).and.returnValue(mockTask);
+      } else {
+        spyOn(ref, putMethod);
+      }
+      task = storage['$' + putMethod](file);
```

Comment:
In the case of `isPutString`, let's actually pass `"raw"` as the second parameter after `fileOrRawString`.

Let's also pass some file metadata for both `put()` and `putString()` to make sure that is getting carried on as well:

```
var metadata = {
  contentType: 'image/jpeg',
};

if (isPutString) {
  task = storage.putString(fileOrRawString, 'raw', metadata);
} else {
  task = storage.put(fileOrRawString, metadata);
}
```

Then you'll need to update the tests below which check the arguments passed to `put()` and `putString()`.